tardis/io/tests/test_HDFWriter.py

```diff
@@ -0,0 +1,138 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pandas.util.testing as pdt
+import pytest
+from astropy import units as u
+from astropy.tests.helper import assert_quantity_allclose
+from numpy.testing import assert_almost_equal, assert_array_almost_equal
+
+from tardis.io.util import HDFWriter
+
+
+#Test Cases
+
+#DataFrame
+#None
+#Numpy Arrays
+#Strings
+#Numeric Values
+#Pandas Series Object
+#MultiIndex Object
+#Quantity Objects with - Numeric Values, Numpy Arrays, DataFrame, Pandas Series, None objects
+
+class MockHDF(HDFWriter, object):
+    hdf_properties = ['property']
+    class_properties = {}
+
+    def __init__(self, property):
+        self.property = property
+
+simple_objects = [1.5, 'random_string', 4.2e7]
+
+@pytest.mark.parametrize("attr", simple_objects)
+def test_simple_write(tmpdir, attr):
+    fname = str(tmpdir.mkdir('data').join('test.hdf'))
+    actual = MockHDF(attr)
+    actual.to_hdf(fname, path='test')
+    expected = pd.read_hdf(fname, key='/test/mock_hdf/scalars')['property']
+    assert actual.property == expected
+
+mock_df = pd.DataFrame({'one': pd.Series([1., 2., 3.], index=['a', 'b', 'c']),
+                        'two': pd.Series([1., 2., 3., 4.], index=['a', 'b', 'c', 'd'])})
+complex_objects = [np.array([4.0e14, 2, 2e14, 27.5]),
+                   pd.Series([1., 2., 3.]), mock_df]
+
+@pytest.mark.parametrize("attr", complex_objects)
+def test_complex_obj_write(tmpdir, attr):
+    fname = str(tmpdir.mkdir('data').join('test.hdf'))
+    actual = MockHDF(attr)
+    actual.to_hdf(fname, path='test')
+    expected = pd.read_hdf(fname, key='/test/mock_hdf/property').values
+    assert_array_almost_equal(actual.property, expected)
+
+arrays = [['L1', 'L1', 'L2', 'L2', 'L3', 'L3', 'L4', 'L4'],
+          ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two']]
+tuples = list(zip(*arrays))
```

Comment:
I think this is a very complicated way to create these arrays. I'd suggest you use `np.array(...)` and then simply transpose it.